### PR TITLE
PLT-274: Datasource for ippool.

### PR DESCRIPTION
**Pull request by nikchern**

### CODE REVIEW

File: docs/resources/macro.md [hunk 1]
```diff
@@ -3,12 +3,12 @@
 page_title: "spectrocloud_macro Resource - terraform-provider-spectrocloud"
 subcategory: ""
 description: |-
-  A resource for creating and managing service output variables and macros.
+  A resource for creating and managing service output variables  and macros.
```
Comment (TylerGillson) [inline docs/resources/macro.md:6]: This change should not be included

File: docs/resources/macro.md [hunk 2]
```diff
@@ -3,12 +3,12 @@
...
 ---
 
 # spectrocloud_macro (Resource)
 
-A resource for creating and managing service output variables and macros.
+A resource for creating and managing service output variables  and macros.
```
Comment (TylerGillson) [inline docs/resources/macro.md:11]: should not be included

File: pkg/client/private_cloud_gateway.go [hunk 3]
```diff
@@ -20,25 +21,48 @@ func (h *V1Client) CreateIpPool(pcgUID string, pool *models.V1IPPoolInputEntity)
...
+	pools, err := h.GetIpPools(pcgUID)
+	if err != nil {
+		return nil, err
+	}
+	for _, pool := range pools {
+		if pool.Metadata.UID == poolUID {
+			return pool, nil
+		}
+	}
+	return nil, nil
```
Comment (TylerGillson) [inline pkg/client/private_cloud_gateway.go:35]: should be returning an error here, but you aren't

**Review by TylerGillson — CHANGES_REQUESTED** [review]

File: pkg/client/private_cloud_gateway.go (hunk 3 above)
Comment (nikchern) [reply pkg/client/private_cloud_gateway.go:35]: thank you for checking, actually already fixed it locally but still verifying on the env.

File: docs/resources/macro.md (hunk 2 above)
Comment (nikchern) [reply docs/resources/macro.md]: reverted

File: docs/resources/macro.md (hunk 1 above)
Comment (nikchern) [reply docs/resources/macro.md]: reverted

**Review by SivaanandM — APPROVED** [review]

**Review by TylerGillson — APPROVED** [review]

**Pull request merged**